Tasks/PublishTestResults/PublishResultsThroughExe.ts

```diff
@@ -0,0 +1,111 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as tl from 'vsts-task-lib/task';
+import * as tr from 'vsts-task-lib/toolrunner';
+import * as vsts from 'vso-node-api';
+
+export class TestResultsPublisher {
+    constructor(matchingTestResultsFiles: string[], mergeResults: string, platform: string, config: string,
+        testRunTitle: string, publishRunAttachments: string, testRunner: string) {
+
+        this.matchingTestResultsFiles = matchingTestResultsFiles;
+        this.mergeResults = mergeResults;
+        this.platform = platform;
+        this.config = config;
+        this.testRunTitle = testRunTitle;
+        this.publishRunAttachments = publishRunAttachments;
+        this.testRunner = testRunner;
+    }
+
+    public async publishResultsThroughExe() {
+        let testResultsPublisherTool: tr.ToolRunner = tl.tool(this.getTestResultsPublisherLocation());
+        let envVars: { [key: string]: string; } = this.getEnvironmentVariables();
+        let args: string[] = this.getArguments();
+        testResultsPublisherTool.arg(args);
+
+        try {
+            await testResultsPublisherTool.exec(<tr.IExecOptions>{ env: envVars });
+        } catch (err) {
+            tl.warning(tl.loc("ErrorTestResultsPublisher", err));
+        }
+    }
+
+    private getTestResultsPublisherLocation(): string {
+        return path.join(__dirname, 'modules/TestResultsPublisher.exe');
+    }
+
+    private getArguments(): string[] {
+        let responseFilePath = this.createResponseFile();
+        // Adding '@' because this is a response file argument
+        let args = ['@' + responseFilePath];
+        return args;
+
+    }
+
+    private createResponseFile(): string {
+        let responseFilePath: string = path.join(__dirname, 'tempResponseFile.txt');
+
+        // Adding quotes around matching file names
+        this.matchingTestResultsFiles = this.modifyMatchingFileName(this.matchingTestResultsFiles);
```

Comment:
no need to pass as param, its already class member

Reply:
Thanks. Done.